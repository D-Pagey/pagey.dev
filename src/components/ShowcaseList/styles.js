import styled from 'styled-components/macro';

export const Wrapper = styled.div`
    margin: 2rem 0;
    width: 100%;
`;
