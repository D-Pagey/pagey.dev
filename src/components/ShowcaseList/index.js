import React from 'react';
import ShowcaseItem from '../ShowcaseItem';
import BarChart from '../BarChart';
import ScatterPlot from '../ScatterPlot';
import * as S from './styles';

const ShowcaseList = () => (
    <S.Wrapper>
        <ShowcaseItem description={['This is a scatter plot']} title="Scatter Plot" thumbnail={<ScatterPlot />} />
        <ShowcaseItem
            description={[
                'Here is a sample of my recent gym session durations in minutes.',
                'This bar chart was built with raw SVGs and only using the scale functions from the D3 library.'
            ]}
            title="Gym Session Duration"
            thumbnail={<BarChart data={[58, 67, 58, 63, 63, 62, 52, 53]} xAxisLabel="Index" yAxisLabel="Minutes" />}
        />
    </S.Wrapper>
);

export default ShowcaseList;
