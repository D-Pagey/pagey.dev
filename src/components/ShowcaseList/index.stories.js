import React from 'react';
import { storiesOf } from '@storybook/react';
import ShowcaseList from '.';

const props = {};

storiesOf('ShowcaseList', module).add('default', () => <ShowcaseList {...props} />);
