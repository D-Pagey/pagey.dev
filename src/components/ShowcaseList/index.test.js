import React from 'react';
import ShowcaseList from '.';

const props = {};

describe('ShowcaseList', () => {
    it('should render', () => {
        const { container } = render(<ShowcaseList {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
