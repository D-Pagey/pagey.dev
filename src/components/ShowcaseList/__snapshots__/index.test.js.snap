// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowcaseList should render 1`] = `
.c1 {
  border-top: 1px solid #f3f3f3;
  display: grid;
  grid-row-gap: 1rem;
  grid-template-columns: 1fr;
  padding: 1rem 0;
}

.c2 {
  margin: 2rem 0 0;
}

.c3 {
  margin: 0 0 1rem;
}

.c4 {
  grid-row: 1;
  max-width: 400px;
}

.c6 {
  max-width: 500px;
}

.c7 {
  fill: skyblue;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c7:hover {
  fill: navy;
}

.c5 {
  max-width: 500px;
}

.c0 {
  margin: 2rem 0 1rem;
  width: 100%;
}

@media (min-width:700px) {
  .c1 {
    grid-column-gap: 1rem;
    grid-template-columns: 1fr 400px;
    grid-template-rows: max-content;
    padding: 1rem 0.5rem;
  }
}

@media (min-width:700px) {
  .c4 {
    grid-column: 2;
    grid-row: 1 / 3;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h4
      class="c2"
    >
      Scatter Plot
    </h4>
    <div>
      <p
        class="c3"
      >
        This is a fictitious scatter plot representing coffee consumption vs hours of sleep.
      </p>
      <p
        class="c3"
      >
        At some point, I would like to actually track these data points and see if there is indeed a negative correlation.
      </p>
    </div>
    <div
      class="c4"
    >
      <svg
        class="c5"
        viewBox="0 0 500 400"
      >
        <line
          stroke="grey"
          stroke-width="1"
          x1="50"
          x2="50"
          y1="365"
          y2="35"
        />
        <text
          text-anchor="end"
          x="45"
          y="40"
        >
          10
        </text>
        <text
          text-anchor="end"
          x="454"
          y="381"
        >
          10
        </text>
        <text
          transform="rotate(270) translate(-225, -180)"
          x="-25"
          y="200"
        >
          Cups of coffee ‚òïÔ∏è Ô∏è
        </text>
        <text
          text-anchor="end"
          x="45"
          y="370"
        >
          0
        </text>
        <text
          x="190"
          y="395"
        >
          Hours of sleep üò¥
        </text>
        <line
          stroke="grey"
          stroke-width="1"
          x1="50"
          x2="450"
          y1="365"
          y2="365"
        />
        <g>
          <circle
            cx="50"
            cy="365"
            data-testid="scatterPlotPoint0"
            fill="green"
            r="9"
          />
        </g>
        <g>
          <circle
            cx="50"
            cy="365"
            data-testid="scatterPlotPoint1"
            fill="green"
            r="9"
          />
        </g>
        <g>
          <circle
            cx="50"
            cy="365"
            data-testid="scatterPlotPoint2"
            fill="green"
            r="9"
          />
        </g>
        <g>
          <circle
            cx="50"
            cy="365"
            data-testid="scatterPlotPoint3"
            fill="green"
            r="9"
          />
        </g>
        <g>
          <circle
            cx="50"
            cy="365"
            data-testid="scatterPlotPoint4"
            fill="green"
            r="9"
          />
        </g>
        <g>
          <circle
            cx="50"
            cy="365"
            data-testid="scatterPlotPoint5"
            fill="green"
            r="9"
          />
        </g>
        <g>
          <circle
            cx="50"
            cy="365"
            data-testid="scatterPlotPoint6"
            fill="green"
            r="9"
          />
        </g>
        <g>
          <circle
            cx="50"
            cy="365"
            data-testid="scatterPlotPoint7"
            fill="green"
            r="9"
          />
        </g>
        <g>
          <circle
            cx="50"
            cy="365"
            data-testid="scatterPlotPoint8"
            fill="green"
            r="9"
          />
        </g>
      </svg>
    </div>
  </div>
  <div
    class="c1"
  >
    <h4
      class="c2"
    >
      Gym Session Duration
    </h4>
    <div>
      <p
        class="c3"
      >
        Here is a sample of my recent gym session durations in minutes.
      </p>
      <p
        class="c3"
      >
        This bar chart was built with raw SVGs and only using the scale functions from the D3 library.
      </p>
    </div>
    <div
      class="c4"
    >
      <svg
        class="c6"
        viewBox="0 0 500 400"
      >
        <text
          x="420"
          y="176.66666666666666"
        >
          Average
        </text>
        <line
          stroke="green"
          stroke-dasharray="2"
          stroke-width="1"
          x1="50"
          x2="480"
          y1="181.66666666666666"
          y2="181.66666666666666"
        />
        <line
          stroke="grey"
          stroke-width="1"
          x1="50"
          x2="50"
          y1="35"
          y2="365"
        />
        <text
          text-anchor="end"
          x="45"
          y="40"
        >
          68
        </text>
        <text
          text-anchor="end"
          x="45"
          y="186.66666666666666"
        >
          60
        </text>
        <text
          transform="rotate(270) translate(-225, -180)"
          x="0"
          y="200"
        >
          Minutes
        </text>
        <text
          text-anchor="end"
          x="45"
          y="370"
        >
          50
        </text>
        <text
          x="250"
          y="398"
        >
          Index
        </text>
        <line
          stroke="grey"
          stroke-width="1"
          x1="50"
          x2="480"
          y1="365"
          y2="365"
        />
        <g>
          <rect
            class="c7"
            data-testid="barChart"
            height="0"
            rx="3"
            width="30"
            x="50"
            y="365"
          />
          <text
            x="58"
            y="380"
          >
            0
          </text>
        </g>
        <g>
          <rect
            class="c7"
            data-testid="barChart"
            height="0"
            rx="3"
            width="30"
            x="107.14285714285714"
            y="365"
          />
          <text
            x="115.14285714285714"
            y="380"
          >
            1
          </text>
        </g>
        <g>
          <rect
            class="c7"
            data-testid="barChart"
            height="0"
            rx="3"
            width="30"
            x="164.28571428571428"
            y="365"
          />
          <text
            x="172.28571428571428"
            y="380"
          >
            2
          </text>
        </g>
        <g>
          <rect
            class="c7"
            data-testid="barChart"
            height="0"
            rx="3"
            width="30"
            x="221.42857142857142"
            y="365"
          />
          <text
            x="229.42857142857142"
            y="380"
          >
            3
          </text>
        </g>
        <g>
          <rect
            class="c7"
            data-testid="barChart"
            height="0"
            rx="3"
            width="30"
            x="278.57142857142856"
            y="365"
          />
          <text
            x="286.57142857142856"
            y="380"
          >
            4
          </text>
        </g>
        <g>
          <rect
            class="c7"
            data-testid="barChart"
            height="0"
            rx="3"
            width="30"
            x="335.7142857142857"
            y="365"
          />
          <text
            x="343.7142857142857"
            y="380"
          >
            5
          </text>
        </g>
        <g>
          <rect
            class="c7"
            data-testid="barChart"
            height="0"
            rx="3"
            width="30"
            x="392.85714285714283"
            y="365"
          />
          <text
            x="400.85714285714283"
            y="380"
          >
            6
          </text>
        </g>
        <g>
          <rect
            class="c7"
            data-testid="barChart"
            height="0"
            rx="3"
            width="30"
            x="450"
            y="365"
          />
          <text
            x="458"
            y="380"
          >
            7
          </text>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
